### DISCUSSION TITLE
Addons scoped to an instance of Marked

### DISCUSSION THREAD

**Discussion by RoenLie** [markedjs/marked#2804] [General]
It might have been like this prior to version 5, but after upgrading I have noticed that any extension added by using the marked.use method, will keep on adding the same extension again and again, without any way to scoping that extension to a one time use, or an instance of marked.

Say for example I want to change my configuration in certain scenarios, I do not seem to be able to do this.

Is there any existing way, or a interest in creating a way we can avoid any extensions added through marked.use to affect all other instances of marked? Or as a more blunt force approach, a way to clear any existing hooks, walkTokens etc that have been put on?

**Comment by UziTech** [comment]
The current way to reset the options is with `marked.setOptions(marked.getDefaults());` This will reset all options including setting the `extensions` option to null effectively removing all extensions.

Creating a marked instance has been on the TODO list for a while. You can see #907 for the discussion.
